addons/portal/controllers/mail.py

```diff
@@ -112,6 +112,17 @@ def _portal_post_has_content(self, res_model, res_id, message, attachment_ids=No
         """ Tells if we can effectively post on the model based on content. """
         return bool(message) or bool(attachment_ids)
 
+    @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
+    def portal_avatar(self, res_id=None, height=50, width=50, access_token=None):
+        """ Get the avatar image in the chatter of the portal """
+        if access_token and res_id:
+            mail_record = request.env['mail.message'].browse(res_id)
+            access_as_sudo = _check_special_access(mail_record.model, mail_record.res_id, token=access_token)
+            record = mail_record if access_as_sudo else request.env.ref('web.image_placeholder').sudo()
```

Comment:
Means `record` is potentially undefined if there is no token or res_id. Would do this, while adding a sudo to mail_record
```py
record_sudo = None
if access_token and res_id:
     record_sudo = mail_record.sudo()
if not record_sudo:
     record_sudo = request.env.ref('web.image_placeholder').sudo()
```